### ISSUE TITLE
distinct on get_all()/between()

### ISSUE THREAD

**Issue by nviennot** [rethinkdb/rethinkdb#3346]
`r.between(:index => :some_index).distinct` does not work. RethinkDB complains that the `id` index and the `some_index` index are both being used. However, `get_all()` works fine. Example:

``` ruby
#!/usr/bin/env ruby
require 'rethinkdb'
include RethinkDB::Shortcuts

r.connect(:host => 'localhost', :port => 28015, :db => :test).repl
r.db_drop(:test).run rescue nil
r.db_create(:test).run
r.table_create(:table).run

r.table(:table).index_create(:field1).run

# returns 0
puts r.table(:table).get_all(1,3, :index => :field1).distinct.count.run

# raises Cannot use 2 indexes in the same operation. Trying to use field1 and id (RethinkDB::RqlRuntimeError)
puts r.table(:table).between(1,3, :index => :field1).distinct.count.run
```

**Comment by AtnNn** [comment]
@nviennot A fix has been released in RethinkDB 1.15.3

**Comment by nviennot** [comment]
nice :+1:
